src/test/java/rx/internal/operators/SingleOnSubscribeUsingTest.java

```diff
@@ -0,0 +1,492 @@
+/**
+ * Copyright 2014 Netflix, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package rx.internal.operators;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+import java.util.*;
+
+import org.junit.Test;
+import org.mockito.InOrder;
+
+import rx.*;
+import rx.Observer;
+import rx.exceptions.TestException;
+import rx.functions.*;
+import rx.subscriptions.Subscriptions;
+
+public class SingleOnSubscribeUsingTest {
+
+    private interface Resource {
+        String getTextFromWeb();
+        
+        void dispose();
+    }
+
+    private static class DisposeAction implements Action1<Resource> {
+
+        @Override
+        public void call(Resource r) {
+            r.dispose();
+        }
+
+    }
+
+    private final Action1<Subscription> disposeSubscription = new Action1<Subscription>() {
+
+        @Override
+        public void call(Subscription s) {
+            s.unsubscribe();
+        }
+
+    };
+
+    @Test
+    public void nonEagerly() {
+        performTestUsing(false);
+    }
+
+    @Test
+    public void eagerly() {
+        performTestUsing(true);
+    }
+
+    private void performTestUsing(boolean disposeEagerly) {
+        final Resource resource = mock(Resource.class);
+        when(resource.getTextFromWeb()).thenReturn("Hello world!");
+
+        Func0<Resource> resourceFactory = new Func0<Resource>() {
+            @Override
+            public Resource call() {
+                return resource;
+            }
+        };
+
+        Func1<Resource, Single<String>> observableFactory = new Func1<Resource, Single<String>>() {
+            @Override
+            public Single<String> call(Resource resource) {
+                return Single.just(resource.getTextFromWeb());
+            }
+        };
+
+        @SuppressWarnings("unchecked")
+        Observer<String> observer = mock(Observer.class);
+        Single<String> observable = Single.using(resourceFactory, observableFactory,
+                new DisposeAction(), disposeEagerly);
+        observable.subscribe(observer);
+
+        InOrder inOrder = inOrder(observer);
+        inOrder.verify(observer).onNext("Hello world!");
+        inOrder.verify(observer).onCompleted();
+        inOrder.verifyNoMoreInteractions();
+
+        // The resouce should be closed
+        verify(resource).dispose();
+    }
+
+    @Test
+    public void withSubscribingTwice() {
+        performTestUsingWithSubscribingTwice(false);
```

Comment:
nit: probably worth to create constants with meaningful name to use them instead of `true/false`, something like:

```java
@Test
public void withSubscribingTwice() {
  performTestUsingWithSubscribingTwice(DONT_DISPOSE_EAGERLY);
}
```